src/styles/utils/_functions.scss

```diff
@@ -0,0 +1,13 @@
+@each $number, $day in $name_days {
+  .calendar--start-day-#{$day} :first-child {
+    margin-left: calc(#{$number} * #{$field-width});
+  }
+}
+
+@for $i from 28 through 31 {
+  .calendar--month-lenght-#{$i} {
+    &:nth-child(n + #{$i + 1}) {
```

Comment:
Changing the number of days in your solution doesn't work as it should. I think this is the right place you should change in order to fix it. Maybe point to calendar__field?